com.unity.render-pipelines.universal/Runtime/Passes/GBufferPass.cs

```diff
@@ -106,7 +106,32 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
 
                 NativeArray<ShaderTagId> tagValues = new NativeArray<ShaderTagId>(m_ShaderTagValues, Allocator.Temp);
                 NativeArray<RenderStateBlock> stateBlocks = new NativeArray<RenderStateBlock>(m_RenderStateBlocks, Allocator.Temp);
-                context.DrawRenderers(renderingData.cullResults, ref drawingSettings, ref m_FilteringSettings, universalMaterialTypeTag, false, tagValues, stateBlocks);
+
+                var activeDebugHandler = GetActiveDebugHandler(renderingData);
+                if (activeDebugHandler != null)
+                {
+                    foreach (DebugRenderSetup debugRenderSetup in activeDebugHandler.CreateDebugRenderSetupEnumerable(context, gbufferCommands))
+                    {
+                        DrawingSettings debugDrawingSettings = debugRenderSetup.CreateDrawingSettings(ref renderingData, drawingSettings);
+
+                        if (debugRenderSetup.GetRenderStateBlock(out RenderStateBlock renderStateBlock))
+                        {
+                            for (int i = 0; i < stateBlocks.Length; i++)
+                                stateBlocks[i] = renderStateBlock;
+
+                            context.DrawRenderers(renderingData.cullResults, ref debugDrawingSettings, ref m_FilteringSettings, universalMaterialTypeTag, false, tagValues, stateBlocks);
+                        }
+                        else
+                        {
+                            context.DrawRenderers(renderingData.cullResults, ref debugDrawingSettings, ref m_FilteringSettings, universalMaterialTypeTag, false, tagValues, stateBlocks);
+                        }
+                    }
+                }
+                else
+                {
+                    context.DrawRenderers(renderingData.cullResults, ref drawingSettings, ref m_FilteringSettings, universalMaterialTypeTag, false, tagValues, stateBlocks);
+                }
```

Comment:
This code is repeated in some places, also users will have to write this if they want to write custom pass that supports debug mode. Maybe we should add helper function to encapsulate this logic?

Reply:
This code is now removed from GBufferPass (since I've opted to use forward renderer for majority of debug drawing). The point about encapsulating is still valid, and I've now addressed it here: https://github.com/Unity-Technologies/Graphics/pull/4089/commits/a3abbc41d8e699ffc895b5b6fa883bf40cadfc98